provider/src/main/kotlin/org/gradle/kotlin/dsl/support/KotlinCompiler.kt

```diff
@@ -272,6 +278,16 @@ class LoggingMessageCollector(
             else -> log.debug { taggedMsg() }
         }
     }
+
+    /**
+     * Safely record a script compilation error.
+     * There'are no guarantee that `MessageCollector`s are not invoked concurrently.
+     */
```

Comment:
👍  Thanks for looking into this and resolving it.

Reply:
Bikeshed: I might have used a concurrent data structure like a `CopyOnWriteArrayList` instead, but this works fine as well.